grammar.js

```diff
@@ -1,489 +1,653 @@
-				field('left', $._expression),
-				field('operator', operator),
-				field('right', $._expression),
-			))
-		)),
-
-		unary_expression: $ => choice(...unary_operators.map(operator => prec(PREC.unary, seq(
-				field('operator', token(operator)),
-				field('operand', $._expression),
-		)))),
-
-		_primary_expression: $ => choice(
-			$._operand,
-			$.index_expression,
-			$.selector_expression,
-			$.call_expression,
-		),
-
-		_operand: $ => choice(
-			$.identifier,
-			$._literal,
-			seq('(', $._expression, ')'),
-		),
-
-		_call_operand: $ => choice(
-			$.builtin,
-			$.qualified_identifier,
-		),
-
-		index_expression: $ => seq($._primary_expression, $.index),
-
-		selector_expression: $ =>	seq($._primary_expression, $.selector),
-
-		call_expression: $ => seq($._call_operand, $.arguments),
-
-		selector: $ => seq($.dot, choice($.identifier, $.simple_string_lit)),
-
-		index: $ => seq(token.immediate('['), $._expression, ']'),
-
-		arguments: $ => seq(
-			'(', $.argument, repeat(seq(',', $.argument)), ')'
-		),
-		argument: $ => $._expression,
-
-		_literal: $ => choice(
-			$.struct_lit,
-			$.list_lit,
-			$._string_lit,
-			$.string_type,
-			$.bytes_type,
-			$.number_type,
-			$.int_lit,
-			$.int_type,
-			$.uint_type,
-			$.float_lit,
-			$.float_type,
-			$.null,
-			$.true,
-			$.false,
-			$.bool_type,
-			$.top_lit,
-			$.bottom_lit,
-		),
-
-		comment: $ => token(choice(
-			seq('//', /.*/),
-		))
-
-	}
+  name: 'cue',
+
+  extras: $ => [
+    $.comment,
+    /\s/,
+  ],
+
+  externals: $ => [
+    $._multi_str_content,
+    $._multi_bytes_content,
+    $._raw_str_content,
+    $._raw_bytes_content,
+    $._multi_raw_str_content,
+    $._multi_raw_bytes_content,
+  ],
+
+  inline: $ => [
+    $.keyword_identifier,
+  ],
+
+  supertypes: $ => [
+    $.expression,
+    $.primary_expression,
+  ],
+
+  conflicts: $ => [
+    [$._embedding, $._label_alias_expr],
+  ],
+
+  word: $ => $.identifier,
+
+  rules: {
+    source_file: $ => seq(
+      optional($.attribute),
+      optional($.package_clause),
+      optional(repeat($.import_declaration)),
+      optionalCommaSep($._declaration),
+    ),
+
+    _package_identifier: $ => alias($.identifier, $.package_identifier),
+
+    package_clause: $ => seq(
+      'package',
+      $._package_identifier,
+    ),
+
+    import_declaration: $ => seq(
+      'import',
+      choice(
+        $.import_spec,
+        $.import_spec_list,
+      ),
+    ),
+
+    import_spec: $ => seq(
+      optional(field('name', choice(
+        '.',
+        '_',
+        $._package_identifier,
+      ))),
+      field('path', $.string),
+    ),
+
+    import_spec_list: $ => seq(
+      '(',
+      optionalCommaSep($.import_spec),
+      ')',
+    ),
+
+    identifier: _ => token(choice(
+      seq(
+        optional(choice('_#', '#', '_')),
+        choice(/\p{L}/, '$', '_'),
+        repeat(choice(/\p{L}/, '$', '_', /[0-9]/)),
+      ),
+      choice('_#', '#', '_'),
```

Comment:
Per the spec `_` is not a valid identifier (and it doesn't evaluate).

`#` and `_#` evaluate fine but I'm not sure it's supposed to be used, I'll have to ask

Reply:
Ah, yes the test runs fine omitting `_`. Let me know about the others..

Reply:
Right!